aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1777,7 +1777,7 @@ void linalg_qr_out_helper(const Tensor& input, const Tensor& Q, const Tensor& R,
   orgqr_stub(input.device().type(), const_cast<Tensor&>(Q), tau);
 }
 
-std::tuple<Tensor, Tensor> _linalg_qr_helper_cpu(const Tensor& input, std::string mode) {
+std::tuple<Tensor, Tensor> _linalg_qr_helper_default(const Tensor& input, std::string mode) {
```

Comment:
Why "default" and not "cpu"?

Reply:
We have now `linalg_qr_helper_magma` that uses MAGMA for the QR decomposition, it can't be implemented using `geqrf_stub` + `orgqr_stub`, because `orgqr_stub` only supports cuSOLVER for CUDA inputs. In addition, MAGMA doesn't follow LAPACK API for `geqrf` and `orgqr` operations that together form the QR decomposition. That's why we need to have a separate function for MAGMA.

And we have `_linalg_qr_helper_default` with "_default" and not "_cpu" because this function supports both CPU and CUDA inputs, for CUDA inputs cuSOLVER&cuBLAS is used.